pkg/ipamd/ipamd.go

```diff
@@ -391,10 +393,10 @@ func New(rawK8SClient client.Client, cachedK8SClient client.Client) (*IPAMContex
 	c.enablePrefixDelegation = usePrefixDelegation()
 	c.enableIPv4 = isIPv4Enabled()
 	c.enableIPv6 = isIPv6Enabled()
+	c.disableENIProvisioning = disableENIProvisioning()
+	disableLeakedENICleanup := c.disableENIProvisioning || disableLeakedENICleanup()
```

Comment:
Nit: Maybe lets find another variable name - `disableLeakedENICleanup` internally now holds `disableENIProvisioning`...

Reply:
I decided to move the `disableENIProvisioning` check into the function, the same as the IPv6 check